src/components/CustomCollection/CustomBookmark/CustomBookmarkForm.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect } from 'react'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import {
+  Form,
+  ButtonGroup,
+  Button,
+  Label,
+  TextInput,
+  Alert,
+  ErrorMessage,
+} from '@trussworks/react-uswds'
+
+type CustomBookmarkFormProps = {
+  onSave: (url: string, label: string) => void
+  onCancel: () => void
+  onDelete?: () => void
+  label?: string
+  url?: string
+  showAddWarning?: boolean
+  nameInputRef: React.RefObject<HTMLInputElement>
+  urlInputRef: React.RefObject<HTMLInputElement>
+}
+export const CustomBookmarkForm = ({
+  onSave,
+  onCancel,
+  onDelete,
+  label,
+  url,
+  showAddWarning,
+  nameInputRef,
+  urlInputRef,
+}: CustomBookmarkFormProps) => {
+  // #TODO: Integrate Formik into our forms following the
+  // wrapper component pattern used in other Truss projects
+
+  const formik = useFormik({
+    initialValues: {
+      // We have to provide label and url as possible initial values
+      // in case a user is editing
+      bookmarkLabel: label || '',
+      bookmarkUrl: url || '',
+    },
+    validateOnChange: false,
+    validateOnBlur: true,
+    enableReinitialize: true,
+    onSubmit: () => {
+      onSave(formik.values.bookmarkUrl, formik.values.bookmarkLabel)
+      formik.resetForm()
+    },
+
+    validationSchema: yup.object({
+      bookmarkLabel: yup.string().required('Link name is required'),
+      bookmarkUrl: yup
+        .string()
+        .transform((_, value) => {
+          // If the user enters a URL without a scheme,
+          // add http:// so we can validate it.
+          // This doesn't affect the value of the input.
```

Comment:
context: yup can't/won't change the actual value in formik, but it can transform it for validation. here, we're adding the scheme if there isn't one so we can take advantage of yup's url validation. if it's invalid, we can show the user an error message before going any further.